text/0079-cdk-2.0.md

```diff
@@ -0,0 +1,337 @@
+
+1. Build and validate package.json - This verifies that the `monocdk-experiment`'s package.json references all of the modules under the `@aws-cdk/` namespace. If not, it will add those packages to the list of dependencies.
+2. Create build directory - Create a temporary directory will code will be staged for build.
+3. Generate `index.ts` - Loop through all directories within the `packages/@aws-cdk` directory of the repository and copy all source code to the buildd directory. Rexport the contents of the copied module from `index.ts` under a namespace based on the modules directory name.
+4. Generate module `package.json` - Create the package.json for the build of `monocdk-experiment`. This has no dependencies and includes `constructs` and `@types/node` as devDependencies.
+5. Install dependencies - Run `npm install`.
+6. Compile with jsii
+7. Stage for local consumption - copies the compiled `.js` files from the build directory back to the repository in the `packages/monocdk-experiment/staging` directory. This allows local modules that depend on `monocdk-experiment` to build against it.
+8. Package - module is packaged as a normal jsii package using `jsii-pacmak` in the repositories packaging step using `pack.sh`.
+
+All `@aws-cdk/` namespaced libraries will continue to be published as normal.
+
+### 2. Pre-release
+
+`monocdk-experiment` is renamed to `aws-cdk-lib`. The version of this module now becomes v2.0 and is tagged as a "pre-release" version. This has varying formats for different language targets.
+
+#### NodeJS
+version number: `2.0.0-beta.1`
+deploying: `npm publish --tag beta`
+installing: `npm install aws-cdk-lib@beta`
+package.json:
+```sh
+"dependencies": {
+  "aws-cdk-lib": "~2.0.0-beta.1"
+}
+```
+**Note: `~` ranges match all versions with the same major, minor, and patch numbers with a greater prerelease number. For example, the above would match `2.0.0-beta.2`. but not `2.0.1-beta.2`.**
+
+#### Java
+version number: `2.0.0-beta-1`
+pom.xml:
+```xml
+<dependencies>
+    <dependency>
+        <groupId>software.amazon.awscdk</groupId>
+        <artifactId>lib</artifactId>
+        <version>2.0.0-beta-1</version>
+    </dependency>
+</dependencies>
+```
+
+#### Dotnet
+version number: `2.0.0-beta.1`
+.csproj:
+```xml
+  <ItemGroup>
+    <PackageReference Include="Amazon.CDK.Lib" Version="2.0.0-beta.1" />
+  </ItemGroup>
+```
+
+#### Python
+version numbeer: `2.0b1`
+requirements.txt: 
+```
+aws-cdk.lib==2.0b1
+```
+
+All `@aws-cdk/` namespaced libraries will continue to be published with current `1.x.x` versions.
+
+### Stable
+
+After a period of testing and gathering feedback from users, `aws-cdk-lib` will be promoted to a stable release with the version `2.0.0`. All other modules, excluding those under the `@aws-cdk/` namespace, will also get promoted to v2.0, though will remain backwards compatible with `@aws-cdk/` modules.
```

Comment:
> All other modules, [...] will also get promoted to v2.0, though will remain backwards compatible with `@aws-cdk/` modules.

I would like to see a table with a list of modules (collapsing all the service packages into one), and what treatment they'll be getting. I'm afraid a couple of these things won't be like the others.